bothan-core/src/manager/crypto_asset_info/price/tasks.rs

```diff
@@ -19,17 +18,13 @@ use crate::monitoring::types::{
 
 // TODO: Allow records to be Option<T>
 /// Computes the price states for a list of signal ids.
-pub async fn get_signal_price_states<S, W>(
+pub async fn get_signal_price_states<S: Store>(
     ids: Vec<String>,
-    workers: &HashMap<String, W>,
+    store: &S,
     registry: &Registry<Valid>,
     stale_cutoff: i64,
     records: &mut Vec<PriceSignalComputationRecord>,
-) -> Vec<PriceState>
-where
-    S: Store + 'static,
-    W: AssetWorker<S>,
-{
+) -> Vec<PriceState> {
     let mut cache = PriceCache::new();
 
     let mut queue = VecDeque::from(ids.clone());
```

Comment:
Just curious why we need deque for this? why not just use Vec (from my understanding you just need stack for prereq)

Reply:
vecdeque was used in the previous iteration, will change to vec as its no longer needed